client/js/components/snippets/snippet-form-dialog.jsx

```diff
@@ -31,16 +37,32 @@ export default class SnippetFormDialog extends React.Component {
       content: this.state.content,
       description: this.state.description,
       language: this.state.language
+    }).then(() => {
```

Comment:
We can always rerender dialog form with that wrong data
but, it could be anoing that someone clicks ```submit``` and waits some time until dialog is closed.